openeo/rest/connection.py

```diff
@@ -1355,3 +1356,37 @@ def paginate(con: Connection, url: str, params: dict = None, callback: Callable
         page += 1
         params = {}
 
+
+
+#
+# For caching connections
+#
+
+_default_url = "openeo-dev.vito.be"
+
+
+def set_backend(url: str):
+    _default_url = url
+
+
+@lru_cache(maxsize=None)
+def _cached_connection(url: str) -> Connection:
+    return openeo.connect(url).authenticate_oidc()
+
+
+def cached_connection(url: Optional[str] = _default_url) -> Connection:
+    """
+    Returns an authenticated openEO connection.
+    Connects to openEO platform by default, but others can be used as well by specifying the url.
+
+    @param url:
+    @return:
+    """
+    return _cached_connection(url)
+
+
+# TODO: I think we can drop these partial functions. We can still define them in openeo-classification if we want.
+#   That is probably a more appropriate place for them.
+terrascope_dev = partial(cached_connection, "openeo-dev.vito.be")
+openeo_platform = partial(cached_connection, "openeo.cloud")
+creo = partial(cached_connection, "openeo-dev.creo.vito.be")
```

Comment:
same as above: these specific back-end urls don't belong in this general connection.py module

Reply:
This code has been removed.